cli/chaosli/cmd/create.go

```diff
@@ -6,29 +6,618 @@
+	kinds := []string{"dns", "network", "cpu", "disk", "node failure"}
+	helpText := `The DNS disruption allows for overriding the A or CNAME records returned by DNS queries.
+The Network disruption allows for injecting a variety of different network issues into your target.
+The CPU and Disk Pressure disruptions apply cpu or IO pressure to your target, respectively.
+Tne Node Failure disruption can either shutdown or restart the targeted node, or the node hosting the targeted pod.
+
+Select one for more information on it.`
+
+	query := initial
+
+	for {
+		response, err := selectInput(query, kinds, helpText)
+		if err != nil {
+			return err
+		}
+
+		switch response {
+		case "..":
+			return nil
+		case "dns":
+			spec.DNS = getDNS()
+
+			if spec.DNS == nil {
+				continue
+			}
+
+			err := spec.DNS.Validate()
+			if err != nil {
+				fmt.Printf("There were some problems with your DNS disruption's spec: %v\n\n", err)
+
+				spec.DNS = nil
+
+				continue
+			}
+		case "network":
+			spec.Network = getNetwork()
+
+			if spec.Network == nil {
+				continue
+			}
+
+			err := spec.Network.Validate()
+			if err != nil {
+				fmt.Printf("There were some problems with your network disruption's spec: %v\n\n", err)
+
+				spec.Network = nil
+
+				continue
+			}
+		case "cpu":
+			spec.CPUPressure = getCPUPressure()
+
+			if spec.CPUPressure == nil {
+				continue
+			}
+
+			err := spec.CPUPressure.Validate()
+			if err != nil {
+				fmt.Printf("There were some problems with your CPU pressure disruption's spec: %v\n\n", err)
+
+				spec.CPUPressure = nil
+
+				continue
+			}
+		case "disk":
+			spec.DiskPressure = getDiskPressure()
+
+			if spec.DiskPressure == nil {
+				continue
+			}
+
+			err := spec.DiskPressure.Validate()
+			if err != nil {
+				fmt.Printf("There were some problems with your disk pressure disruption's spec: %v\n\n", err)
+
+				spec.DiskPressure = nil
+
+				continue
+			}
+		case "node failure":
+			spec.NodeFailure = getNodeFailure()
+
+			if spec.NodeFailure == nil {
+				continue
+			}
+
+			err := spec.NodeFailure.Validate()
+			if err != nil {
+				fmt.Printf("There were some problems with your node failure disruption's spec: %v\n\n", err)
+
+				spec.NodeFailure = nil
+
+				continue
+			}
+		}
+
+		i := indexOfString(kinds, response)
+		kinds = append(kinds[:i], kinds[i+1:]...)
+
+		if query == initial {
+			kinds = append(kinds, "..")
+		}
+
+		query = followUp
+	}
+}
+
+func confirmKind(kind string, helpText string) bool {
+	return confirmOption(fmt.Sprintf("Would you like to include the disruption %s?", kind), helpText)
+}
+
+func confirmOption(query string, helpText string) bool {
+	var result bool
+
+	prompt := &survey.Confirm{
+		Message: query,
+		Help:    helpText,
+	}
+
+	err := survey.AskOne(prompt, &result)
+
+	if err == terminal.InterruptErr {
+		os.Exit(1)
```

Comment:
Just a thought; I know that the exit code can be changed to help better identify locations of errors. Wonder if it would be helpful in this situation to use different exit codes depending where the exit occurs.

Reply:
If this line were in the controller, 100%, but absolutely no one should ever be using this feature outside of a human-CLI interaction, so there's no need to specially identify where we exited. It may be useful to separate _why_ we exited, and use a different exit code for ctrl+c vs. a "real" error, but as long as it's just a ctrl+c, any positive exit code (as long as we're consistent), should be good

Reply:
Oh great point, yes that makes a lot of sense, thanks for explaining.